bee-ledger/src/whiteflag/mod.rs

```diff
@@ -33,41 +35,22 @@ pub fn init(
     index: u32,
     state: LedgerState,
     coo_config: ProtocolCoordinatorConfig,
-    bee_node: Arc<BeeNode>,
+    node_builder: NodeBuilder<BeeNode>,
     bus: Arc<Bus<'static>>,
-    shutdown: &mut Shutdown,
-) -> mpsc::UnboundedSender<LedgerWorkerEvent> {
```

Comment:
I'm a bit weirded out by this function. Is this a permanent thing?

Reply:
The reasoning is that I would like not to publicly expose all of the workers all the time so I prefer to go through an init function. Does that make sense ?

Reply:
That makes sense. I wonder whether some sort of `with_default_workers` method on `NodeBuilder` would be useful here?

Reply:
No sure to understand what it would do

Reply:
Have the `with_default_workers` initiate these workers instead. It's not important at this stage though, feel free to ignore.